### PR TITLE
[KAT-2415] Adding test for shared memory CDLP app

### PR DESCRIPTION
**Pull request by amazloumi**
This PR is aimed to add different tests for the CDLP app in the shared memory environment:
	1	Test cdlp using some generated small graph using TopologyGeneration.h.
	2	Test cdlp in python API
	3	Test cdlp-cpu
	◦	adding a add_test_scale in lonestar/analytics/cpu/cdlp/CMakeLists.txt

//following will be added later
	◦	adding end-to-end third-party test: It uses OpenG(GraphBig)+Graphalytics as the third-party platform to generate an output file and uses that output file in CI to compare and test CDLP katana output. This will be useful after implementing the distributed CDLP as well.

### CODE REVIEW

**Comment by gurbinder533** [comment]
@amazloumi Are you still working on it?

**Comment by amazloumi** [comment]
> @amazloumi Are you still working on it?

This part is ready to review. The only thing I want to add here next is the end-to-end third-party output file test.

**Review by uditagar — APPROVED** [review]
looks fine

File: lonestar/analytics/cpu/cdlp/CMakeLists.txt [hunk 1]
```diff
@@ -2,3 +2,5 @@ add_executable(cdlp-cpu cdlp_cli.cpp)
 add_dependencies(apps cdlp-cpu)
 target_link_libraries(cdlp-cpu PRIVATE Katana::galois lonestar)
 
+add_test_scale(small cdlp-cpu NO_VERIFY INPUT rmat15 INPUT_URI "${RDG_RMAT15}" --algo=Synchronous)
```
Comment (gurbinder533) [inline lonestar/analytics/cpu/cdlp/CMakeLists.txt:5]: All all the algorithm for testing here. Is there Asynhronous as well?

**Review by gurbinder533 — APPROVED** [review]
Looks good.

File: lonestar/analytics/cpu/cdlp/CMakeLists.txt (hunk 1 above)
Comment (amazloumi) [reply lonestar/analytics/cpu/cdlp/CMakeLists.txt:5]: Asynchronous has not been implemented yet.

Comment (amazloumi) [reply lonestar/analytics/cpu/cdlp/CMakeLists.txt:5]: the tests for asynchronous will be added once the asynchronous gets implemented.

**Pull request merged**